### PR TITLE
fix the issue that inconsistent highlighting occurs when multiple mentions are used without spaces

### PR DESCRIPTION
**Pull request by agilejune**
### Fixed Issues
$ https://github.com/Expensify/App/issues/21753
Proposal: https://github.com/Expensify/App/issues/21753#issuecomment-1759589159

### Tests
1. Go to any chat.
2. Type the following cases
```
@test.example+2@gmail.com@here
@here@test.example+2@gmail.com
@test.example+2@gmail.com@test.example+2@gmail.com
@here@here@gmail.com
@here@gmail.com@here
@here@here+1@gmail.com
@here+1@gmail.com@here
```
3. Observe that all the mentions are highlighted as expected.

### QA
Same as tests step

### Offline tests
Same as tests step

### Screenshots/Videos
<details>
<summary>Web</summary>

image

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by agilejune** [comment]
**_I have read the CLA Document and I hereby sign the CLA_**

**Comment by amyevans** [comment]
@aimane-chnaif can you comment here so I can assign you for review please?

File: __tests__/ExpensiMark-HTML-test.js [hunk 1]
```diff
@@ -1298,6 +1298,36 @@ test('Test for @here mention without space or supported styling character', () =
     expect(parser.replace(testString)).toBe(resultString);
 });
 
+test('Test user mention and here mention, which are concatenated without space', () => {
```
[the remark this answers is not in the document]
Comment (agilejune) [reply __tests__/ExpensiMark-HTML-test.js:1301]: It's updated with general ones

[the remark this answers is not in the document]
Comment (agilejune) [reply __tests__/ExpensiMark-HTML-test.js:1301]: done as well

**Comment by agilejune** [comment]
@aimane-chnaif good catch of edge case
I will fix to cover all those cases

**Comment by agilejune** [comment]
image

**Comment by agilejune** [comment]
@aimane-chnaif please check again

**Comment by puneetlath** [comment]
@agilejune looks like you have some failing tests.

**Comment by agilejune** [comment]
@puneetlath @aimane-chnaif 
all done

File: lib/ExpensiMark.js [hunk 2]
```diff
@@ -202,7 +208,7 @@ export default class ExpensiMark {
                     return this.modifyTextForQuote(regex, textToProcess, replacement);
                 },
                 replacement: (g1) => {
-                    const replacedText = this.replace(g1, {filterRules: ['heading1'], shouldEscapeText: false});
+                    const replacedText = this.replace(g1, { filterRules: ['heading1'], shouldEscapeText: false });
```
[the remark this answers is not in the document]
Comment (agilejune) [reply lib/ExpensiMark.js]: oops, sorry about it.
all restored
@aimane-chnaif

File: lib/str.js [hunk 3]
```diff
@@ -944,7 +944,8 @@ const Str = {
      */
     isValidMention(mention) {
         // Mentions can start @ proceeded by a space, eg "ping @user@domain.tld"
-        if (/[\s@]/g.test(mention.charAt(0))) {
+        // or by a @here, eg "@here@user@domain.tld"
+        if (/[\s@]/g.test(mention.charAt(0)) || /^@here.*/g.test(mention)) {
```
[the remark this answers is not in the document]
Comment (agilejune) [reply lib/str.js]: @aimane-chnaif you're right. it's needless
I removed it

**Review by puneetlath — DISMISSED** [review]

**Comment by puneetlath** [comment]
Sorry for the delay. Code looks good to me, but tests are failing.

**Comment by agilejune** [comment]
@puneetlath It's fixed

**Review by puneetlath — APPROVED** [review]
